test/src/tests-session-manager.ts

```diff
@@ -0,0 +1,607 @@
+            clock.tick(30 * 60000);
+            expect(endSessionSpy.called).to.equal(true);
+
+            // Let test know async process is done
+            done();
+        });
+
+        it('should set a global timer', () => {
+            mParticle.init(apiKey, window.mParticle.config);
+            const mpInstance = mParticle.getInstance();
+
+            // Manually clear global timer since it gets set during init
+            // window.clearTimeout(mpInstance._Store.globalTimer);
+            mpInstance._Store.globalTimer = null;
+            expect(mpInstance._Store.globalTimer).to.eq(null);
+
+            mpInstance._SessionManager.setSessionTimer();
+            expect(mpInstance._Store.globalTimer).to.be.ok;
+        });
+    });
+
+    describe('#resetSessionTimer', () => {
+        it('should call startNewSession if sessionId is not defined', () => {
+            mParticle.init(apiKey, window.mParticle.config);
+            const mpInstance = mParticle.getInstance();
+
+            mpInstance._Store.sessionId = null;
+
+            const startNewSessionSpy = sinon.spy(
+                mpInstance._SessionManager,
+                'startNewSession'
+            );
+
+            mpInstance._SessionManager.resetSessionTimer();
+            expect(startNewSessionSpy.called).to.equal(true);
+        });
+
+        it('should NOT call startNewSession if sessionId is defined', () => {
+            mParticle.init(apiKey, window.mParticle.config);
+            const mpInstance = mParticle.getInstance();
+
+            mpInstance._Store.sessionId = 'test-session-id';
+
+            const startNewSessionSpy = sinon.spy(
+                mpInstance._SessionManager,
+                'startNewSession'
+            );
+
+            mpInstance._SessionManager.resetSessionTimer();
+            expect(startNewSessionSpy.called).to.equal(false);
+        });
+
+        it('should NOT restart a session if webviewBridgeEnabled is true', () => {
+            mParticle.init(apiKey, window.mParticle.config);
+            const mpInstance = mParticle.getInstance();
+
+            mpInstance._Store.webviewBridgeEnabled = true;
+
+            const startNewSessionSpy = sinon.spy(
+                mpInstance._SessionManager,
+                'startNewSession'
+            );
+            const clearSessionTimeoutSpy = sinon.spy(
+                mpInstance._SessionManager,
+                'clearSessionTimeout'
+            );
+            const setSessionTimerSpy = sinon.spy(
+                mpInstance._SessionManager,
+                'setSessionTimer'
+            );
+
+            mpInstance._SessionManager.resetSessionTimer();
+            expect(startNewSessionSpy.called).to.equal(false);
+            expect(clearSessionTimeoutSpy.called).to.equal(false);
+            expect(setSessionTimerSpy.called).to.equal(false);
+        });
+
+        it('should reset session timer', () => {
```

Comment:
this is a bit generic of a test description.  Should it be `should call clearSessionTimeout and setSessionTimer`?